### PR TITLE
MVT loader fixes

### PR DESCRIPTION
**Pull request by jesusbotella**
Here is the pull request to fix tests and add some other things that I missed in the previous one.

Now, by default, loader returns an array of [Point](https://github.com/mapbox/point-geometry) arrays using `mapbox/vector-tile-js` without transforming them at any point in our loader.

If you want to get an array of GeoJSON features, you should include `geojson` property and `_tileProperties` so that the library projects the points correctly. If `geojson` is set to true and `_tileProperties` is missing, the geometry data won't be correct.

I think this PR addresses all feedback provided in https://github.com/uber-web/loaders.gl/pull/625. Let me know your thoughts :)

### CODE REVIEW

**Comment by jesusbotella** [comment]
Tests are passing in my local environment but fail in CI 🤔. I don't know if rerunning them will make them pass, as other non-related tests fail as well.

**Comment by xintongxia** [comment]
> Tests are passing in my local environment but fail in CI 🤔. I don't know if rerunning them will make them pass, as other non-related tests fail as well.

I think the the result files can't be loaded by `import someData from 'some-json-file'` directly, but it can work with the following way as `loaders.gl` use alias in all the tests in all the modules. 

try 
```
// Geometry Array Results
import decodedPolygonsGeometry from '@loaders.gl/mvt/test/results/decoded_mvt_polygons_array.json';

// GeoJSON Results
import decodedPointsGeoJSON from '@loaders.gl/mvt/test/results/decoded_mvt_points.json';
import decodedLinesGeoJSON from '@loaders.gl/mvt/test/results/decoded_mvt_lines.json';
import decodedPolygonsGeoJSON from '@loaders.gl/mvt/test/results/decoded_mvt_polygons.json';
```

Or you can just use `const decodedPointsGeoJSON = './results/decoded_mvt_points.json';`
But the first option is preferred as we would like to avoid relative path.

File: modules/mvt/src/mvt-loader.js [hunk 1]
```diff
@@ -12,8 +12,7 @@ export const MVTWorkerLoader = {
   category: 'geometry',
   options: {
     mvt: {
-      geojson: true,
-      tileProperties: {},
```
Comment (ibgreen) [inline modules/mvt/src/mvt-loader.js]: Nit: Why no default for `_tileProperties`?

Comment (jesusbotella) [reply modules/mvt/src/mvt-loader.js]: I thought that as there was no `geojson` property, `tileProperties` were useless, but I think it is good to keep that default value as well.

**Comment by jesusbotella** [comment]
I have just renamed `_tileProperties` to `tileIndex`, as I think it is clearer and more understandable.

I have just added options documentation to the Markdown file as well. I wrote a little description about both properties, let me know if you think it is understandable enough or not :)

File: modules/mvt/docs/api-reference/mapbox-vector-tile-loader.md [hunk 2]
```diff
@@ -16,12 +16,32 @@ Loader for the [Mapbox Vector Tile](https://docs.mapbox.com/vector-tiles/specifi
...
+| --------------- | ------------------------------------------------ | ------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `mvt.geojson`   | Boolean                                          | true         | If `true`, parser will return a flat array of GeoJSON representations of the features with coordinates decoded from provided tile index. If `false`, parser will return a flat array of [Point](https://github.com/mapbox/point-geometry) objects with feature coordinates decoded from (0, 0). |
+| `mvt.tileIndex` | Object ({x: `number`, y: `number`, z: `number`}) | _No default_ | Mandatory if using `geojson` output (It will return null coordinates if omitted). An object containing tile index values (x, y, z) to decode coordinates.                                                                                                                                       |
```
Comment (ibgreen) [inline modules/mvt/docs/api-reference/mapbox-vector-tile-loader.md]: Nit: Not sure that returning `null` is the best choice. Ideas: 
- Set `tileIndex` to some default, e.g. (`0, 0, 0`)
- throw an error (or at least emit a warning) if `geojson` is set but `tileIndex` is omitted.

(We could also combine `geojson` and `tileIndex` but I am thinking that all the geospatial loaders should support a `geojson` option so rather keep that as a simple boolean for consistency)

File: modules/mvt/test/mvt-loader.spec.js [hunk 3]
```diff
@@ -93,7 +93,7 @@ test('Polygons MVT to GeoJSON', async t => {
   const loaderOptions = {
     mvt: {
       geojson: true,
-      _tileProperties: {
+      tileIndex: {
```
Comment (ibgreen) [inline modules/mvt/test/mvt-loader.spec.js]: Nits:
- Add test cases for bad option combos (geojson: true, no tileIndex, bad tileIndex).
- If throwing, I recommend adding `t.throws(() => func)` test cases.

**Comment by ibgreen** [comment]
@jesusbotella  I will land this. Since the CI is failing, we may to disable some test case until we get that sorted 

(it could be some subtle issue with worker loader etc so maybe best we look at it).

**Comment by ibgreen** [comment]
Published as `2.1.0-alpha.2`

**Pull request merged**